src/lib/djerba/plugins/pwgs/analysis/plugin.py

```diff
@@ -51,31 +53,39 @@ def configure(self, config):
 
     def extract(self, config):
         wrapper = self.get_config_wrapper(config)
-        mrdetect_results = pwgs_tools.preprocess_results(self, config[self.identifier][pc.RESULTS_FILE])
-        hbc_results = self.preprocess_hbc(config[self.identifier][pc.HBC_FILE])
-        reads_detected = self.preprocess_vaf(config[self.identifier][pc.VAF_FILE])
-        pwgs_base64 = self.write_pwgs_plot(config[self.identifier][pc.HBC_FILE], 
-                                           config[self.identifier][pc.VAF_FILE],
-                                           output_dir = self.workspace.print_location())
+        mrdetect_results = pwgs_tools.preprocess_results(self, wrapper.get_my_string(pc.RESULTS_FILE))
+        hbc_results = self.preprocess_hbc(wrapper.get_my_string(pc.HBC_FILE))
+        reads_detected = self.preprocess_vaf(wrapper.get_my_string(pc.VAF_FILE))
+        pwgs_base64 = self.write_pwgs_plot(wrapper.get_my_string(pc.HBC_FILE), wrapper.get_my_string(pc.VAF_FILE), output_dir=self.workspace.print_location())
         self.logger.info("PWGS ANALYSIS: Finished preprocessing files")
-        data = self.get_starting_plugin_data(wrapper, self.PLUGIN_VERSION)       
-        results =  {
-                pc.CTDNA_OUTCOME: mrdetect_results[pc.CTDNA_OUTCOME],
-                pc.SIGNIFICANCE: mrdetect_results[pc.SIGNIFICANCE],
-                pc.TUMOUR_FRACTION_READS: float('%.1E' % Decimal( reads_detected*100 / hbc_results[pc.READS_CHECKED] )),
-                pc.SITES_CHECKED: hbc_results[pc.SITES_CHECKED],
-                pc.READS_CHECKED: hbc_results[pc.READS_CHECKED],
-                pc.SITES_DETECTED: hbc_results[pc.SITES_DETECTED],
-                pc.READS_DETECTED: reads_detected,
-                pc.PVALUE: mrdetect_results[pc.PVALUE],
-                pc.COHORT_N: hbc_results[pc.COHORT_N],
-                'pwgs_base64': pwgs_base64,
-                'files': {
-                    'results_file': config[self.identifier][pc.RESULTS_FILE],
-                    'hbc_results': config[self.identifier][pc.HBC_FILE],
-                    'vaf_results': config[self.identifier][pc.VAF_FILE]
-                }
+        data = self.get_starting_plugin_data(wrapper, self.PLUGIN_VERSION)
+        workspace_dir = self.workspace.get_work_dir()
+        # Read from the case_overview JSON file and populate results
+        with open(os.path.join(workspace_dir, "pWGS_case_overview_output.json"), 'r') as json_file:
```

Comment:
This adds a dependency on the PWGS case overview plugin. This is fine, but needs to be handled more carefully.
- The plugin tests are currently failing, because `pWGS_case_overview_output.json` is not available when the tests are run
- If `pWGS_case_overview_output.json` does not exist, the plugin crashes without applying the "not found" default values you've defined

Basically you have 2 possible choices:
1. Require  `pWGS_case_overview_output.json` to be present. Get rid of the default values (which don't really make sense in that situation), and add a copy of the file to the test data so tests can run successfully. We usually put files like this in the `djerba_test_data` Bitbucket repository and copy them to the test directory at runtime, I can give you some tips if needed.
2. Make  `pWGS_case_overview_output.json` optional. In this case the plugin should check whether it exists, and apply default values if the file is not found.

I'm OK with either option, but the second one will be easier to implement. Let me know what you think and if you have any questions.